modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleFreeContainer.java

```diff
@@ -0,0 +1,221 @@
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Future;
+
+/**
+ * Testcontainers implementation for Oracle.
+ * <p>
+ * Supported image: {@code container-registry.oracle.com/database/free}
+ * <p>
+ * Exposed ports: 1521
+ * <p>
+ * MIT-licensesd by Bernd Eckenfels, based on OracleContainer from oracle-xe.
+ */
+public class OracleFreeContainer extends JdbcDatabaseContainer<OracleFreeContainer> {
+
+    public static final String NAME = "oracle";
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("container-registry.oracle.com/database/free");
+
+    // https://container-registry.oracle.com/ords/f?p=113:4:16246212162327:::RP,4:P4_REPOSITORY,AI_REPOSITORY,P4_REPOSITORY_NAME,AI_REPOSITORY_NAME:1863,1863,Oracle%20Database%20Free,Oracle%20Database%20Free&cs=3gQONqcVhKMn1mUNRMJjdHgofFIzDG97WTyRx3tHBfkK8axlWmSmJRNF_YnrAizYZZj5kyAnyS9O_nudXM3Xw6w
+    static final String DEFAULT_TAG = "23.3.0.0";
+
+    static final String IMAGE = DEFAULT_IMAGE_NAME.getUnversionedPart();
+
+    static final int ORACLE_PORT = 1521;
+
+    private static final int DEFAULT_STARTUP_TIMEOUT_SECONDS = 240;
+
+    private static final int DEFAULT_CONNECT_TIMEOUT_SECONDS = 120;
+
+    // Container defaults
+    static final String DEFAULT_DATABASE_NAME = "freepdb1";
+
+    static final String DEFAULT_SID = "free";
+
+    static final String DEFAULT_SYSTEM_USER = "system";
+
+    static final String DEFAULT_SYS_USER = "sys";
+
+    // Test container defaults
+    static final String APP_USER = "test";
+
+    static final String APP_USER_PASSWORD = "test";
+
+    // Restricted user and database names
+    private static final List<String> ORACLE_SYSTEM_USERS = Arrays.asList(DEFAULT_SYSTEM_USER, DEFAULT_SYS_USER);
+
+    private String databaseName = DEFAULT_DATABASE_NAME;
+
+    private String username = APP_USER;
+
+    private String password = APP_USER_PASSWORD;
+
+    private boolean usingSid = false;
+
+    /**
+     * @deprecated use {@link #OracleContainer(DockerImageName)} instead
+     */
+    @Deprecated
+    public OracleFreeContainer() {
+        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG));
+    }
+
+    public OracleFreeContainer(String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    public OracleFreeContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        preconfigure();
+    }
+
+    public OracleFreeContainer(Future<String> dockerImageName) {
+        super(dockerImageName);
+        preconfigure();
+    }
+
+    private void preconfigure() {
```

Comment:
There is already a `configure()` method that can be overridden so you lines 83 and 88 can be deleted.

Reply:
Yes maybe, I was sticking to the orig8nal source, should we fix/change the original class then as well?

Reply:
Not as part of this PR. Another PR would be welcomed.